src/tests/ascent/t_ascent_mpi_unique_ids.cpp

```diff
@@ -0,0 +1,166 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) Lawrence Livermore National Security, LLC and other Ascent
+// Project developers. See top-level LICENSE AND COPYRIGHT files for dates and
+// other details. No copyright assignment is required to contribute to Ascent.
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+//-----------------------------------------------------------------------------
+///
+/// file: t_ascent_partition.cpp
+///
+//-----------------------------------------------------------------------------
+
+
+#include "gtest/gtest.h"
+
+#include <ascent.hpp>
+#include <ascent_logging.hpp>
+
+#include <iostream>
+#include <math.h>
+#include <mpi.h>
+
+#include <conduit_blueprint.hpp>
+#include <conduit_blueprint_mpi_mesh_examples.hpp>
+#include <conduit_relay.hpp>
+
+#include "t_config.hpp"
+#include "t_utils.hpp"
+
+
+using namespace std;
+using namespace conduit;
+using namespace ascent;
+
+//-----------------------------------------------------------------------------
+TEST(ascent_partition, test_indiv_rank_non_unique)
+{
+    Node n;
+    ascent::about(n);
+
+    //
+    // Create an example mesh.
+    //
+    Node data, verify_info;
+
+    //
+    //Set Up MPI
+    //
+    int par_rank;
+    int par_size;
+    MPI_Comm comm = MPI_COMM_WORLD;
+    MPI_Comm_rank(comm, &par_rank);
+    MPI_Comm_size(comm, &par_size);
+
+    // use spiral , with 20 domains
+    conduit::blueprint::mpi::mesh::examples::spiral_round_robin(20,data,comm);
+
+    EXPECT_TRUE(conduit::blueprint::mesh::verify(data,verify_info));
+
+    int root = 0;
+    if(par_rank == root)
+    	ASCENT_INFO("Testing local non unique IDs");
+
+    if(par_rank == root)
+    {
+      int num_domains = data.number_of_children();
+      for(int i = 0; i < num_domains; i++)
+      {
+        conduit::Node &dom = data.child(i);
+	dom["state/domain_id"] = 0;
+      }
+    }
+    
+    //
+    // Run Ascent
+    //
+
+    Ascent ascent;
+
+    Node ascent_opts;
+    ascent_opts["runtime"] = "ascent";
+    ascent_opts["mpi_comm"] = MPI_Comm_c2f(comm);
+    ascent.open(ascent_opts);
+
+    EXPECT_THROW(ascent.publish(data),conduit::Error);
+}
```

Comment:
Lol. It throws an error but doesn't catch it. @cyrush have any guesses? I tested out throwing other errors from EnsureDomains, like consistent IDs and it is also not catching them (https://github.com/Alpine-DAV/ascent/blob/e52b7bb8c9fd131f2fd49edf58037cc5ef77a166/src/libs/ascent/runtimes/ascent_main_runtime.cpp#L554) 
```

[ RUN      ] ascent_partition.test_indiv_rank_non_unique
[Error] Ascent::publish
file: /sw/summit/ums/ums010/2023_01/frontier/debug_build/ascent/src/libs/ascent/runtimes/ascent_main_runtime.cpp
line: 670
message:
Local Domain IDs are not unique on ranks: 0

Testing local non unique IDs/sw/summit/ums/ums010/2023_01/frontier/debug_build/ascent/src/tests/ascent/t_ascent_mpi_unique_ids.cpp:85: Failure
Expected: ascent.publish(data) throws an exception of type conduit::Error.
  Actual: it throws nothing.
[  FAILED  ] ascent_partition.test_indiv_rank_non_unique (1207 ms)
```

Reply:
By default, Ascent will not throw beyond the interface (we don't want to crash client codes).

You need to use:
```
ascent_opts["exceptions"] = "forward";
```

To allow them to be thrown outside of Ascent.

Reply:
Ahh makes sense! Thanks!